### PR TITLE
Create phpcs static check for DiConfigTest

### PR DESCRIPTION
**Pull request by loginesta**
This sniffer looks for occurrences of obsolete nodes: `param`, `instance`, `array`, `item[@key]` and `value`.
Returns a warning for each occurrence found.
https://jira.corp.magento.com/browse/AC-660

### CODE REVIEW

File: Magento2/ruleset.xml [hunk 1]
```diff
@@ -223,6 +223,11 @@
         <severity>8</severity>
         <type>warning</type>
     </rule>
+    <rule ref="Magento2.Legacy.DiConfig">
+        <severity>8</severity>
+        <type>warning</type>
+        <include-pattern>*\/DiConfigUnitTest.xml$</include-pattern>
```
Comment (sivaschenko) [inline Magento2/ruleset.xml:229]: Is this test for di.xml files only?
```suggestion
        <include-pattern>*\/di.xml$</include-pattern>
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @loginesta ! Can you please verify the test functionality running it against the Magento code with changed di.xml files?

File: composer.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
     "require": {
         "php": ">=7.3",
         "squizlabs/php_codesniffer": "^3.6",
-        "webonyx/graphql-php": "^14.9"
+        "webonyx/graphql-php": "^14.9",
+        "ext-simplexml": "*",
+        "ext-dom": "*"
```
Comment (svera) [inline composer.json:15]: You need to update also `composer.lock`

Comment (loginesta) [reply composer.json:15]: forgot to push it 😉  thanks

**Review by sivaschenko — APPROVED** [review]

**Comment by sivaschenko** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by magento-engcom-team** [comment]
@sivaschenko the pull request successfully imported.

**Review by svera — APPROVED** [review]

**Pull request merged**